### PR TITLE
#120: API containers Unordered-Map transition

### PR DESCRIPTION
**Pull request by marcinwrobel1986**
* API/containers/Unordered-Map transition from .md to .rst

before (.md) | after (.rst)
:----------------: | :-------------:
image | image
image | image
image | image

### CODE REVIEW

File: docs/source/API/containers/Unordered-Map.rst [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+.. role:: cppkokkos(code)
+	:language: cppkokkos
+
+
+Kokkos's unordered map is designed to efficiently handle tens of thousands of concurrent insertions.
+Consequently, the API is significantly different from the standard unordered_map.
+The two key differences are *fixed capacity* and *index based*.
+
+- *Fixed capacity*: The capacity of the unordered_map is fix when inside a parallel algorithm.
```
Comment (Rombur) [inline docs/source/API/containers/Unordered-Map.rst:12]:
```suggestion
- *Fixed capacity*: The capacity of the unordered_map is fixed when inside a parallel algorithm.
```

File: docs/source/API/containers/Unordered-Map.rst [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  between memory spaces) the map uses integer indexes. This also allows the map to store data
+  in cache friendly ways. The availability of indexes is managed by an internal atomic bitset based on ``uint32_t``.
+
+
+Description
+-----------
+
+.. cppkokkos:class:: template <typename Key, typename Value, typename Device = Kokkos::DefaultExecutionSpace> UnorderedMap
+
+   :tparam Key: Must be a POD
```
Comment (ajpowelsnl) [inline docs/source/API/containers/Unordered-Map.rst:26]: What is a "POD"?

File: docs/source/API/containers/Unordered-Map.rst [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+      Was the key/value pair successfully inserted into the map
+
+   .. cppkokkos:kokkosinlinefunction:: bool existing() const;
+
+      Is the key already present in the map
+
+   .. cppkokkos:kokkosinlinefunction:: bool failed() const;
+
+      Did the insert fail?
```
Comment (ajpowelsnl) [inline docs/source/API/containers/Unordered-Map.rst:111]: Is an error emitted upon failure, or does a changed boolean just cruise on by, and no one is the wiser?

**Review by ajpowelsnl — APPROVED** [review]
Please move towards standardizing API entries:  function name, Header File, Description, Usage and Example

File: docs/source/API/containers/Unordered-Map.rst [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+- *Fixed capacity*: The capacity of the unordered_map is fix when inside a parallel algorithm.
+  This means that an insert can fail when the capacity of the map is exceeded.
+  The capacity of the map can be changed (rehash) from the host.
+
+- *Index based*: Instead of returning pointers or iterators (which would not work when moving
+  between memory spaces) the map uses integer indexes. This also allows the map to store data
+  in cache friendly ways. The availability of indexes is managed by an internal atomic bitset based on ``uint32_t``.
+
+
+Description
```
[the remark this answers is not in the document]
Comment (fnrizzi) [reply docs/source/API/containers/Unordered-Map.rst:21]: we can do this separately as a follow up

**Pull request merged**